simdjson-flags.cmake

```diff
@@ -35,39 +35,40 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 # Flags used by exes and by the simdjson library (project-wide flags)
 #
 add_library(simdjson-flags INTERFACE)
+add_library(simdjson-internal-flags INTERFACE)
 if(MSVC)
-  target_compile_options(simdjson-flags INTERFACE /nologo /D_CRT_SECURE_NO_WARNINGS)
-  target_compile_options(simdjson-flags INTERFACE /WX /W3 /sdl)
+  target_compile_options(simdjson-internal-flags INTERFACE /nologo /D_CRT_SECURE_NO_WARNINGS)
+  target_compile_options(simdjson-internal-flags INTERFACE /WX /W3 /sdl)
 else()
-  target_compile_options(simdjson-flags INTERFACE -fPIC)
+  target_compile_options(simdjson-internal-flags INTERFACE -fPIC)
   if (NOT SIMDJSON_GOOGLE_BENCHMARKS) # Google Benchmark can't be compiled without warnings with -Weffc++
-    target_compile_options(simdjson-flags INTERFACE -Weffc++)
+    target_compile_options(simdjson-internal-flags INTERFACE -Weffc++)
   endif()
-  target_compile_options(simdjson-flags INTERFACE -Werror -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self -Wconversion -Wno-sign-conversion)
+  target_compile_options(simdjson-internal-flags INTERFACE -Werror -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self -Wconversion -Wno-sign-conversion)
 endif()
 
 # Optional flags
 option(SIMDJSON_IMPLEMENTATION_HASWELL "Include the haswell implementation" ON)
 if(NOT SIMDJSON_IMPLEMENTATION_HASWELL)
-  target_compile_definitions(simdjson-flags INTERFACE SIMDJSON_IMPLEMENTATION_HASWELL=0)
+  target_compile_definitions(simdjson-internal-flags INTERFACE SIMDJSON_IMPLEMENTATION_HASWELL=0)
```

Comment:
@lemire @furkanusta I'm trying to decide what to do about these. If you do this:

```cmake
set(SIMDJSON_IMPLEMENTATION_HASWELL=OFF)
add_subdirectory(simdjson) # or find_project? I'm not familiar enough
add_executable(blah blah.cpp)
target_link_libraries(blah simdjson)
```

Then do we expect simdjson to compile without HASWELL in it? ... I think we do.

Reply:
(Note: example is aspirational and may or may not compile.)

Reply:
Yeah, this is the only thing I think we need to change--these should be a part of simdjson-flags so they are public interface.

Reply:
If find_package is used in CMake, then we are using an already-built library and it is not possible to deactivate HASWELL etc.

However, it is possible with add_directory

```CMake
set(SIMDJSON_IMPLEMENTATION_HASWELL ON CACHE STRING "" FORCE)
add_subdirectory(simdjson EXCLUDE_FROM_ALL)
add_executable(foo foo.cpp)
target_link_libraries(foo PUBLIC simdjson)
```
This works even now. It still builds the library but now we can control the parameters.

In this case _HASWELL option in CMake is already part of the public API. Users do not need to now which macros this option enables, and making the macros public felt like we are polluting the consumer's project.

Reply:
Grokked. So the rubric is "any flags that affect the headers should be in `simdjson-flags`, but flags affecting only the .cpp should be in `simdjson-internal` in anticipation users will use the library. Works for me.

As an aside, I really appreciate that you push back on suggestions you think are wrong, it is so easy for people to just grumble and accede to requests just to get a patch in.